tests/storage-spec.js

```diff
@@ -0,0 +1,179 @@
+import '../app/scripts/src/main';
+
+describe('languageStorage factory', function () {
+
+    beforeAll(function () {
+
+        this.setupModule = function () {
+
+            angular.mock.module('language-select', ($provide) => {
+                $provide.service('languageSelectConfig', () => {
+                    return {
+                        availableLanguages: () => this.languageChoices,
+                        defaultLanguageId: () => 'se',
+                    };
+                });
+            });
```

Comment:
You can do this simpler by passing an object to `mock.module`. This also avoids confusion with requiring nested arrow-functions so `this` is the spec.
```js
this.mockLanguageId = 'se';
this.mockLanguageChoices = [];

angular.mock.module('language-select');
angular.mock.module({
    languageSelectConfig: {
        availableLanguages: () => this.mockLanguageChoices,
        defaultLanguageId: () => this.mockLanguageId
    }
});
```